diff -uNpr fio-3.28.orig/backend.c fio-3.28/backend.c
--- fio-3.28.orig/backend.c	2021-09-08 10:59:48.000000000 -0400
+++ fio-3.28/backend.c	2021-12-13 14:32:38.009680028 -0500
@@ -1803,10 +1803,12 @@ static void *thread_main(void *data)
 		goto err;
 
 	errno = 0;
+	/*
 	if (nice(o->nice) == -1 && errno != 0) {
 		td_verror(td, errno, "nice");
 		goto err;
 	}
+	*/
 
 	if (o->ioscheduler && switch_ioscheduler(td))
 		goto err;
@@ -2628,6 +2630,7 @@ int fio_backend(struct sk_out *sk_out)
 		sfree(cgroup_list);
 	}
 
+	sync();
 	fio_sem_remove(startup_sem);
 	stat_exit();
 	return exit_value;
diff -uNpr fio-3.28.orig/fio.c fio-3.28/fio.c
--- fio-3.28.orig/fio.c	2021-09-08 10:59:48.000000000 -0400
+++ fio-3.28/fio.c	2021-12-13 14:25:19.515723937 -0500
@@ -26,9 +26,15 @@
 int main(int argc, char *argv[], char *envp[])
 {
 	int ret = 1;
+	char *args[4];
 
 	compiletime_assert(TD_NR <= TD_ENG_FLAG_SHIFT, "TD_ENG_FLAG_SHIFT");
 
+	args[0] = strdup("./fio");
+	args[2] = strdup("/mnt/rand-rw.fio");
+	args[1] = strdup("--output=/mnt/results");
+	args[3] = NULL;
+
 	if (initialize_fio(envp))
 		return 1;
 
@@ -39,7 +45,7 @@ int main(int argc, char *argv[], char *e
 	if (fio_server_create_sk_key())
 		goto done;
 
-	if (parse_options(argc, argv))
+	if (parse_options(3, args))
 		goto done_key;
 
 	/*
diff -uNpr fio-3.28.orig/Makefile fio-3.28/Makefile
--- fio-3.28.orig/Makefile	2021-09-08 10:59:48.000000000 -0400
+++ fio-3.28/Makefile	2021-12-09 12:19:28.995818151 -0500
@@ -21,7 +21,7 @@ endif
 
 DEBUGFLAGS = -DFIO_INC_DEBUG
 CPPFLAGS= -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -DFIO_INTERNAL $(DEBUGFLAGS)
-OPTFLAGS= -g -ffast-math
+OPTFLAGS= -g 
 FIO_CFLAGS= -std=gnu99 -Wwrite-strings -Wall -Wdeclaration-after-statement $(OPTFLAGS) $(EXTFLAGS) $(BUILD_CFLAGS) -I. -I$(SRCDIR)
 LIBS	+= -lm $(EXTLIBS)
 PROGS	= fio
